20181010_Choropleth_Maps_in_R/Customisable Choropleth Maps.Rmd

```diff
@@ -0,0 +1,489 @@
+
+```{r, message=FALSE}
+theme_map <- function() {
+  theme_minimal() +
+    theme(
+      text = element_text(family = "sans"), # Arial font
+      axis.line = element_blank(),
+      axis.text = element_blank(),
+      axis.ticks = element_blank(),
+      axis.title = element_blank(),
+      panel.grid = element_blank(),
+      plot.background = element_blank(), 
+      panel.background = element_blank(), 
+      legend.background = element_blank(),
+      panel.border = element_blank()
+    )
+}
+```
+
+So now we can plot the map using out new theme:
+
+```{r, eval=TRUE}
+map + theme_map()
+```
+
+# Customise your breaks
+
+So now we have a good looking map. However the continuous scale isn't very clear.
+
+We can split the scale up to into discrete categories using the `cut_number()` function. This splits a numeric vector into intervals containing equal number of points. The first argument is the variable we are 'cutting' and the next argument is the number of intervals we want.
+
+```{r, eval=TRUE}
+ggplot() + 
+  geom_polygon(data = map_data, 
+               aes(x = long, y = lat, group = group, fill = cut_number(Percentage, 5)), 
+               col = "black") +
+  coord_equal()+
+  theme_map()
+```
+
+Our colours now change to the default dicrete colour scale but we will look at how to change these a bit later.
+
+If you want an even bigger of choice of breaks or you want to make your own, we can use `classIntervals()` function from the classInt package. This gives us commonly used methods for choosing univariate class intervals for mapping or other graphics purposes. 
+
+The different methods available are: "fixed", "sd", "equal", "pretty", "quantile", "kmeans", "hclust", "bclust", "fisher", or "jenks". We can use `?classIntervals` to get a full description of the different methods.
+
+Here we are going to use 'jenks' breaks. This is a data clustering method designed to determine the best arrangement of values into different classes by reducing the variance within classes and maximising the variance between classes. It's been checked for consistency with ArcView and ArcGIS.
+
+```{r, message=FALSE}
+jenks_breaks <- classIntervals(dummy$Percentage, n=5, style = "jenks")
+```
+
+We can check our breaks like so:
+```{r, eval=TRUE}
+jenks_breaks$brks 
+```
+
+If you're not sure whether the breaks you have created are right for your data, you can compare interval methods to one another. For example, let's have a look at "equal" breaks:
+
+```{r, eval=TRUE}
+equal_breaks <- classIntervals(dummy$Percentage, n=5, style = "equal")
+equal_breaks$brks
+```
+
+We can have a closer look distribution of the variable by plotting a quick histogram of the variable then use `abline()` in different colours to see the breaks we've defined.
+```{r, eval=TRUE}

```

Comment:
I think it would be good to demo how this could be done in ggplot - just to tie it in with other coffee and coding sessions 

```r
vlines <- data.frame(
  jenks = jenks_breaks$brks,
  equal = equal_breaks$brks
) %>% tidyr::gather("method", "brks", jenks:equal)

ggplot(dummy, aes(x=Percentage))+
  geom_histogram(stat = "bin", binwidth = 10)+
  geom_vline(data=vlines, aes(xintercept=brks, color=method))+
  theme_bw()
```

Reply:
done!